src/NuGet.Clients/NuGet.VisualStudio.Common/Verify.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.VisualStudio.Common
+{
+    /// <summary> Argument verification helpers. </summary>
+    internal static class Verify
+    {
+        /// <summary> Throws <see cref="ArgumentNullException"/> if <paramref name="argument"/> is null. </summary>
+        /// <typeparam name="T"> Type of argument. </typeparam>
+        /// <param name="argument"> Argument to verify. </param>
+        /// <param name="argumentName"> Argument name. </param>
+        internal static void ArgumentIsNotNull<T>(T argument, string argumentName)
+        {
+            if (argument == null)
```

Comment:
Honestly, this might be a good utility to have in *all* our classes...I'll steal this idea and add it in a fture PR.

Reply:
Yup. Probably move to NuGet.Common 🙂

Reply:
We use shared compilation, https://github.com/NuGet/NuGet.Client/tree/dev/build/Shared. 
Example: https://github.com/NuGet/NuGet.Client/blob/dev/build/Shared/SharedExtensions.cs

Reply:
Why we are using shared compliation? Are there parts of NuGet which do not reference NuGet.Common?

Reply:
NuGet.Common is part of the SDK (packages that anyone in the world can use to provide NuGet functionality in their own apps), so in my opinion it should only contain public APIs for doing NuGet related things. For this reason I dislike `NuGet.Common` as a concept, as it's not clear what NuGet functionality it's supposed to provide. But I would not be supportive of adding something as generic as this `ArgumentIsNotNull<T>` helper being in the public API.